### DISCUSSION TITLE
Auto-format clang-format with extra args for null ls

### DISCUSSION THREAD

**Discussion by ciuckc** [LazyVim/LazyVim#762] [Q&A] [answered]
Hi is there a way to provide some these arguments to clangd for the auto-formatter?
`-style={BasedOnStyle: Google, IndentWidth: 4, PointerAlignment: Left` ? Thank you very much!

**Comment by amaanq** [comment, marked as answer]
.clang-format file in project root or user root

**Reply by amaanq** [reply]
yes

**Reply by JulesVerne22** [reply]
Is there a way to do this with extra_args instead or via the lua file at all?

**Reply by dpetka2001** [reply]
If you mean the `extra_args` from `null-ls`, then the following should work
```lua
  {
    "jose-elias-alvarez/null-ls.nvim",
    opts = function(_, opts)
      local nls = require("null-ls")
      opts.sources = vim.list_extend(opts.sources, {
        nls.builtins.formatting.clang_format.with({
          extra_args = {
            "-style=file:" .. vim.fn.expand("~/projects/C/.clang-format"),
          },
        }),
      })
    end,
  },
```
But this way it will be globally enforced in every project you want to format with `.clang-format`. If you want per project I think it would be better to just leave it to `lsp` since it seems to respect the settings if it discovers a `.clang-format` file itself. Or instead of the line with `-style=file`, you could just pass the arguments you want directly. Example `--style="{BasedOnStyle: llvm, IndentWidth: 8}"`.

**Reply by amaanq** [reply]
I'd use the lsp over the null-ls source as that installs a separate python program rather than using the one built into the lsp, not sure how to configure it in there though

**Reply by prabanjanraja** [reply]
it was working fine but suddenly it is not formatting the file according to .clang-format that is present in the root directory any one know what the issue is
